horace_core/sqw/@sqw/section.m

```diff
@@ -67,7 +67,34 @@
                                         new_axis, ...
                                         win(n).data.proj);
 
-        wout(n).pix = win(n).pix.get_pix_in_ranges(bl_start,bl_size);
+        if win(n).pix.is_filebacked
+            chunk_size = get(hor_config, 'mem_chunk_size');
+
+            block_chunks = split_data_blocks(bl_start, bl_size, chunk_size);
+            num_chunks = numel(block_chunks);
+
+            wout(n) = wout(n).get_new_handle();
+
+            pix_ind = 1;
+
+            for iter = 1:num_chunks
+                chunk = block_chunks{iter};
+                pix_start = chunk{1};
+                block_sizes = chunk{2};
+
+                data = win(n).pix.get_pix_in_ranges(pix_start, block_sizes, false, false);
+
+                wout(n).pix.format_dump_data(data.data, pix_ind);
```

Comment:
All operations which take partial pixels must recalculate pixel data averages.  For memory backed it happens automatically when your assing new data to `PixedDataMemory` but here it has to be present. 

I hope pix API does everything else correctly

Reply:
Section doesn't change the bins, it merely extracts them, so this is taking whole bins and the bin averages don't need recalculating. The `data_range` is updated in the underlying `DnDBase.section`